godtools/App/Features/ShareTool/ShareTool/ShareToolViewModel.swift

```diff
@@ -36,14 +36,24 @@ class ShareToolViewModel: ShareToolViewModelType {
         )
     }
     
+    private var analyticsScreenName: String {
+        return resource.abbreviation + "-" + String(pageNumber)
+    }
+    
+    private var analyticsSiteSection: String {
+        return resource.abbreviation
+    }
+    
+    private var analyticsSiteSubSection: String {
+        return ""
+    }
+    
     func pageViewed() {
         
-        let analyticsScreenName: String = resource.abbreviation + "-" + String(pageNumber)
-        
-        analytics.pageViewedAnalytics.trackPageView(trackScreen: TrackScreenModel(screenName: analyticsScreenName, siteSection: resource.abbreviation, siteSubSection: ""))
+        analytics.pageViewedAnalytics.trackPageView(trackScreen: TrackScreenModel(screenName: analyticsScreenName, siteSection: analyticsSiteSection, siteSubSection: analyticsSiteSubSection))
                 
-        analytics.trackActionAnalytics.trackAction(trackAction: TrackActionModel(screenName: analyticsScreenName, actionName: AnalyticsConstants.Values.shareIconEngaged, siteSection: resource.abbreviation, siteSubSection: "", url: nil, data: [
-            AnalyticsConstants.ActionNames.shareIconEngagedCountKey: 1
+        analytics.trackActionAnalytics.trackAction(trackAction: TrackActionModel(screenName: analyticsScreenName, actionName: AnalyticsConstants.ActionNames.shareIconEngaged, siteSection: analyticsSiteSection, siteSubSection: analyticsSiteSubSection, url: nil, data: [
+            AnalyticsConstants.Keys.shareAction: 1
```

Comment:
Hey @reldredge71 looks like you fixed the value name for this analytics?

Reply:
@levieggertcru I noticed that `Keys.shareAction` already had the same value as `ActionNames.shareIconEngagedCountKey`, so I just decided to keep one and remove the other.  Plus, I wanted to clean up `AnalyticsConstants` and make it easier to understand; I wanted `ActionNames` to be the names of the `trackAction` events, and I wanted `Keys` to be the keys of the `data` object being passed to the analytics events (basically, everything that Cru tracks and has the "cru." prefix).   I didn't know that `ActionNames.shareIconEngagedCountKey` needed fixing, what should it be?

Reply:
Hey @reldredge71 nothing needs to be fixed, I thought the key was incorrect and the change here was addressing that, but now I see it was duplicated as you said.  This looks good to me.